### ISSUE TITLE
1.19.4 Planter Upgrade Will Destroy Farmland

### ISSUE THREAD

**Issue by HeatSignature** [Alex-the-666/Rats#799]
Type: Female Rat

Upgrades: Undead, Planter, Whitelist (set to Wheat Seeds).

Orders: Command the Rat to pull seeds from a chest in the center of set harvest range then set Rat to Harvest.

Issue: Rat ignores whitelist, plants different random plant seeds, also destroys the farmland it's planting on. When the farmland is destroyed it also manages to plant the seeds on the terracotta blocks underneath.

Currently using the Replanter upgrade as a temp workaround in the meantime :)

**Comment by GizmoTheMoonPig** [comment]
> Rat ignores whitelist, plants different random plant seeds

I'm unable to reproduce this part. I filled a chest with all sorts of different crop seeds and the rat only picks up whats in the whitelist. If the chest didn't contain any wheat seeds the rat did nothing. The only time the rat didnt follow the whitelist was when I made him hold other seeds, but that's intended, as the whitelist is only for picking up/dropping off items. Its very possible I fixed this in a past commit as I had to do a couple other fixes regarding the whitelist/blacklist upgrades so this may not be an issue anymore. 

> destroys the farmland it's planting on. When the farmland is destroyed it also manages to plant the seeds on the terracotta blocks underneath.

This was caused by a major goof in the planting logic, which also uncovered ANOTHER oversight in the logic. I completely forgot that seeds are considered blocks, so sometimes the planting logic would fall back to the block placement logic instead of the seed logic. Because of this, I learned that the block placing logic was not placing the spot it was checking, rather it was placing the block below the targeted position. Luckily this is a very easy fix but still an annoying goof regardless.
